tensorflow/compiler/xrt/xrt_state.cc

```diff
@@ -184,45 +185,43 @@ void XRTTupleAllocation::ReleaseBuffers() {
   // call. To avoid a leak, there must be no error-case returns from here until
   // the end of the method.
   auto shaped_buffer = scoped_buffer->release();
-  *allocation = new XRTTupleAllocation(
-      device_ordinal, backend->memory_allocator(),
-      shaped_buffer.on_host_shape(), shaped_buffer.on_device_shape());
+  *allocation = new XRTTupleAllocation(device_ordinal, allocator,
+                                       shaped_buffer.on_host_shape(),
+                                       shaped_buffer.on_device_shape());
   (*allocation)
-      ->InitializeFromShapedBuffer(shaped_buffer, backend->memory_allocator(),
-                                   device_ordinal);
+      ->InitializeFromShapedBuffer(shaped_buffer, allocator, device_ordinal);
   (*allocation)->SetDeviceMemorySize();
   return Status::OK();
 }
 
 /*static*/ Status XRTTupleAllocation::CreateUninitialized(
     const xla::Shape& shape, XRTMemoryManager* memory_manager,
-    xla::Backend* backend, int device_ordinal,
-    XRTTupleAllocation** allocation) {
+    xla::Backend* backend, int device_ordinal, XRTTupleAllocation** allocation,
+    se::DeviceMemoryAllocator* allocator) {
   std::unique_ptr<xla::ScopedShapedBuffer> scoped_buffer;
-  TF_RETURN_IF_ERROR(AllocateScopedShapedBuffer(
-      memory_manager, backend, device_ordinal, shape, &scoped_buffer));
+  TF_RETURN_IF_ERROR(AllocateScopedShapedBuffer(memory_manager, backend,
+                                                device_ordinal, shape,
+                                                &scoped_buffer, allocator));
 
   // By releasing the ScopedShapedBuffer we ensure that the underlying storage
   // won't be freed when the buffer goes out of scope at the end of this
   // call. To avoid a leak, there must be no error-case returns from here until
   // the end of the method.
   auto shaped_buffer = scoped_buffer->release();
-  *allocation = new XRTTupleAllocation(
-      device_ordinal, backend->memory_allocator(),
-      shaped_buffer.on_host_shape(), shaped_buffer.on_device_shape());
+  *allocation = new XRTTupleAllocation(device_ordinal, allocator,
+                                       shaped_buffer.on_host_shape(),
+                                       shaped_buffer.on_device_shape());
   (*allocation)
-      ->InitializeFromShapedBuffer(shaped_buffer, backend->memory_allocator(),
-                                   device_ordinal);
+      ->InitializeFromShapedBuffer(shaped_buffer, allocator, device_ordinal);
   (*allocation)->SetDeviceMemorySize();
   return Status::OK();
 }
 
 /*static*/ Status XRTTupleAllocation::CreateFromBuffer(
     const xla::ShapedBuffer& shaped_buffer, const xla::Shape& on_host_shape,
     const xla::Shape& on_device_shape, xla::Backend* backend,
```

Comment:
doesn't seems like we need to pass`backend` anymore

Reply:
Now TPU side needs this.